stocks2.lua

```diff
@@ -0,0 +1,544 @@
+            frame_style = gui.INTERIOR_FRAME,
+            subviews = {
+                widgets.EditField({
+                    view_id = "search_box",
+                    frame = {t = 1, l = 0},
+                    label_text = "Search: ",
+                    on_char = function(ch) return ch:match("[%l%s]") end,
+                    on_change = function(new_text, old_text)
+                        current_search_query = new_text
+                        self:filter_list(current_search_query)
+                    end,
+                }),
+                widgets.Label({
+                    text = {{text = "Items: ", pen = COLOR_LIGHTGREEN}},
+                    frame = {t = 3, l = 0},
+                    view_id = "item_count_label"
+                }),
+                widgets.List({
+                    view_id = "item_list",
+                    frame = {t = 4, b = 1, l = 1, r = 1},
+                    text_pen = {fg = COLOR_GREY, bg = COLOR_BLACK},
+                    text_hpen = {fg = COLOR_BLACK, bg = COLOR_GREY},
+                    cursor_pen = {fg = COLOR_LIGHTGREEN, bg = COLOR_DARKGREY},
+                }),
+            },
+        }),
+        widgets.Panel({
+            view_id = "right_panel",
+            frame = {t = 0, b = 0, r = 0, l = 72},
+            frame_style = gui.INTERIOR_FRAME,
+            subviews = {
+                widgets.Label({
+                    text = "Filters (Alt+Key toggles)",
+                    frame = {t = 0, l = 0},
+                    view_id = "filters_header_label",
+                }),
+                widgets.Label({
+                    text = "to be filled in...",
+                    frame = {t = 1, l = 0},
+                    view_id = "filters_label",
+                }),
+                widgets.Label({
+                    text = {
+                        {text = "Shift-C", pen = COLOR_LIGHTGREEN}, {text = ": Clear All", pen = COLOR_WHITE}, NEWLINE,
+                        {text = "Shift-E", pen = COLOR_LIGHTGREEN}, {text = ": Enable All", pen = COLOR_WHITE}, NEWLINE,
+                        {text = "Shift-G", pen = COLOR_LIGHTGREEN}, {text = ": Toggle Grouping", pen = COLOR_WHITE}
+                    },
+                    frame = {t = 8, l = 0},
+                    view_id = "filters_label2",
+                }),
+                widgets.Label({
+                    text = {
+                        {text = "-+", pen = COLOR_LIGHTGREEN}, {text = ": Min Qual: ", pen = COLOR_WHITE}, {text = "tbd", pen = COLOR_BROWN}, NEWLINE,
+                        {text = "/*", pen = COLOR_LIGHTGREEN}, {text = ": Max Qual: ", pen = COLOR_WHITE}, {text = "tbd", pen = COLOR_BROWN}, NEWLINE,
+                        {text = "Shift-W", pen = COLOR_LIGHTGREEN}, {text = ": Min Wear: ", pen = COLOR_WHITE}, {text = "tbd", pen = COLOR_BROWN}
+                    },
+                    frame = {t = 12, l = 0},
+                    view_id = "filters_qual_wear",
+                }),
```

Comment:
This should be a `RangeSlider` (and accompanying `CycleHotkeyLabel`s). Example: https://github.com/DFHack/dfhack/blob/develop/plugins/lua/zone.lua#L169